### PR TITLE
Exclude circe from d4s.

### PR DESCRIPTION
**Pull request by aliubymov-playq**

### CODE REVIEW

File: d4s/src/main/scala/d4s/DynamoClient.scala [hunk 1]
```diff
@@ -17,9 +17,15 @@ object DynamoClient {
     dynamo: DynamoComponent
   ) extends DynamoClient[F] {
 
-    def raw[T](f: DynamoDbClient => T)(implicit notInMonadSanityCheck: T <:!< F[_, _]): F[Throwable, T] = {
+    def raw[T](f: DynamoDbClient => T)(implicit ev: NotMonad[T, F]): F[Throwable, T] = {
```
Comment (vpodlnk) [inline d4s/src/main/scala/d4s/DynamoClient.scala:20]: @aliubymov-playq, IMO but previous variant was better to read and understand, it was like reading a simple condition.

Comment (neko-kai) [reply d4s/src/main/scala/d4s/DynamoClient.scala:20]: It depended on `shapeless.<!:<`, so we either have to copy it or write an alternative. but tbh this sanity check is not crucial, it's not required to keep it if it's in any way inconvenient

Comment (aliubymov-playq) [reply d4s/src/main/scala/d4s/DynamoClient.scala:20]: @vpodlnk @neko-kai I've thought of deleting this but decided to left old behavior. So if you think that this part is not crucial, then I'll delete this check. (Anyway, I suppose that this check a bit useless)

**Comment by aliubymov-playq** [comment]
@neko-kai @vpodlnk It's kind ready for merge. So if you don't mind I'll merge this.

**Comment by aliubymov-playq** [comment]
Also, I've updated `iterable` and `mapLike` decoder to follow fail fast logic instead of traversing across whole collection and returning last one error.

**Pull request merged**